pkg/hubble/exporter/config_watcher_test.go

```diff
@@ -0,0 +1,252 @@
+			assert.Contains(t, err.Error(), tc.expectedErrorMsg)
+		})
+	}
+}
+
+func TestReloadNotificationReceived(t *testing.T) {
+	// given
+	file := createTestConfigFile(t)
+
+	configReceived := false
+
+	// when
+	reloadInterval = 1 * time.Millisecond
+	sut := NewConfigWatcher(file.Name(), func(_ context.Context, _ uint64, config dynamicExportersConfig) {
+		configReceived = true
+	})
+	defer sut.Stop()
+
+	// then
+	assert.Eventually(t, func() bool {
+		return configReceived
+	}, 1*time.Second, 1*time.Millisecond)
+
+}
+
+func createTestConfigFile(t *testing.T) *os.File {
+	testFileContent := `
+flowlogs:
+- name: "test001"
+  filePath: "/var/log/network/flow-log/pa/test001.log"
+  fieldMask: []
+  includeFilters: []
+  excludeFilters: []
+  end: "2023-10-09T23:59:59-07:00"
+- name: "test002"
+  filePath: "/var/log/network/flow-log/pa/test002.log"
+  fieldMask: ["source.namespace", "source.pod_name", "destination.namespace", "destination.pod_name", "verdict"]
+  includeFilters:
+  - source_pod: ["default/"]
+    event_type:
+    - type: 1
+  - destination_pod: ["frontend/nginx-975996d4c-7hhgt"]
+  excludeFilters: []
+  end: "2023-10-09T23:59:59-07:00"
+- name: "test003"
+  filePath: "/var/log/network/flow-log/pa/test003.log"
+  fieldMask: ["source", "destination","verdict"]
+  includeFilters: []
+  excludeFilters:
+  - destination_pod: ["ingress/"]
+`
+	return createConfigFile(t, testFileContent)
+}
+
+func createInvalidYamlTestConfigFile(t *testing.T) *os.File {
+	testFileContent := `
+flowlogs:
+- name: "test001"
+  filePath: "/var/log/network/flow-log/pa/test001.log"
+  fieldMask: "this", "is", "invalid"
+  includeFilters: []
+  excludeFilters: []
+  end: "2023-10-09T23:59:59-07:00"
+`
+	return createConfigFile(t, testFileContent)
+}
+
+func createDuplicatedNameYamlTestConfigFile(t *testing.T) *os.File {
+	testFileContent := `
+flowlogs:
+- name: "test001"
+  filePath: "/var/log/network/flow-log/pa/test001.log"
+  fieldMask: []
+  includeFilters: []
+  excludeFilters: []
+  end: "2023-10-09T23:59:59-07:00"
+- name: "test001"
+  filePath: "/var/log/network/flow-log/pa/test002.log"
+  fieldMask: []
+  includeFilters: []
+  excludeFilters: []
+  end: "2023-10-09T23:59:59-07:00"
+`
+	return createConfigFile(t, testFileContent)
+}
+
+func createDuplicatedPathYamlTestConfigFile(t *testing.T) *os.File {
+	testFileContent := `
+flowlogs:
+- name: "test001"
+  filePath: "/var/log/network/flow-log/pa/test001.log"
+  fieldMask: []
+  includeFilters: []
+  excludeFilters: []
+  end: "2023-10-09T23:59:59-07:00"
+- name: "test002"
+  filePath: "/var/log/network/flow-log/pa/test001.log"
+  fieldMask: []
+  includeFilters: []
+  excludeFilters: []
+  end: "2023-10-09T23:59:59-07:00"
+`
+	return createConfigFile(t, testFileContent)
+}
+
+func createConfigFile(t *testing.T, content string) *os.File {
+	file, err := os.CreateTemp(t.TempDir(), "config.yaml")
+	if err != nil {
+		t.Fatalf("failed creating test file %v", err)
+	}
+
+	file.Write([]byte(content))
```

Comment:
nit, non-blocking: Consider Fatalf on error to make the test fail

Reply:
done, thanks for suggestion